ui/core/components/individual_sim_ui/consumes_picker.ts

```diff
@@ -41,144 +35,122 @@ export class ConsumesPicker extends Component {
-			picker = new IconEnumPicker(
-				elem,
-				this.simUI.player,
-				IconInputs.makeFlasksInput(flaskOptions, 'Flask')
-			);
-		}
-
-		// All current flasks are a level 50+ requirement
-		const updateFlask = () => {
-			if (this.simUI.player.getLevel() >= 50){
-				picker?.restoreValue();
-				ele!.classList.remove('hide');
-			} else {
-				picker?.storeValue();
-				ele!.classList.add('hide');
-			}
-		};
-		this.simUI.player.levelChangeEmitter.on(updateFlask);
-		updateFlask();
+		const rowElem = this.rootElem.appendChild(fragment.children[0] as HTMLElement);
+		const flasksElem = this.rootElem.querySelector('.consumes-flasks') as HTMLElement;
+
+		this.buildPickers({
+			changeEmitters: [this.simUI.player.levelChangeEmitter],
+			containerElem: rowElem,
+			options: [
+				{
+					getConfig: () => ConsumablesInputs.makeFlasksInput(
+						relevantStatOptions(ConsumablesInputs.FLASKS_CONFIG, this.simUI)
+					),
+				}
+			],
+			parentElem: flasksElem,
+		})
 	}
 
 	private buildWeaponImbuePicker() {
 		let fragment = document.createElement('fragment');
 		fragment.innerHTML = `
-    <div class="consumes-row input-root input-inline">
+    	<div class="consumes-row input-root input-inline">
         <label class="form-label">Weapon Imbues</label>
-        <div class="consumes-row-inputs consumes-mainhand"></div>
-    </div>
+        <div class="consumes-row-inputs consumes-weapon-imbues"></div>
+    	</div>
     `;
 
-		this.rootElem.appendChild(fragment.children[0] as HTMLElement);
-
-		const weaponOptions = this.simUI.splitRelevantOptions([
-			{ item: WeaponImbue.BrillianWizardOil, stats: [Stat.StatSpellPower] },
-			{ item: WeaponImbue.BrilliantManaOil, stats: [Stat.StatHealing, Stat.StatSpellPower] },
-			{ item: WeaponImbue.DenseSharpeningStone, stats: [Stat.StatAttackPower] },
-			{ item: WeaponImbue.ElementalSharpeningStone, stats: [Stat.StatAttackPower] },
-			{ item: WeaponImbue.BlackfathomManaOil, stats: [Stat.StatSpellPower, Stat.StatMP5] },
-			{ item: WeaponImbue.BlackfathomSharpeningStone, stats: [Stat.StatMeleeHit] },
-			{ item: WeaponImbue.WildStrikes, stats: [Stat.StatMeleeHit] },
-		]);
-		if (weaponOptions.length) {
-			const elem = this.rootElem.querySelector('.consumes-mainhand') as HTMLElement;
-			new IconEnumPicker(
-				elem,
-				this.simUI.player,	
-				IconInputs.makeMainHandImbuesInput(weaponOptions, 'Weapon Imbues'),
-			);
-		}
+		const rowElem = this.rootElem.appendChild(fragment.children[0] as HTMLElement);
+		const imbuesElem = this.rootElem.querySelector('.consumes-weapon-imbues') as HTMLElement;
+
+		this.buildPickers({
+			changeEmitters: [this.simUI.player.levelChangeEmitter],
+			containerElem: rowElem,
+			options: [
+				{
+					getConfig: () => ConsumablesInputs.makeMainHandImbuesInput(
+						relevantStatOptions(ConsumablesInputs.WEAPON_IMBUES_MH_CONFIG, this.simUI),
+						'Main-Hand',
+					),
+				},
+				{
+					getConfig: () => ConsumablesInputs.makeOffHandImbuesInput(
```

Comment:
Will this be appropriately hidden if no off-hand weapon is equipped?

Reply:
Ah good point!

Reply:
Implemented logic for both mainhand and offhand imbues and added the `gearchangeemitter`. Confirmed that equipping/unequipping maindhands and offhands showed/hid the pickers